enclave-runtime/src/lib.rs

```diff
@@ -891,12 +890,14 @@ where
 	let mut call_hashes = Vec::<H256>::new();
 
 	// load state before executing any calls
-	let mut state = load_initialized_state(&shard)?;
-	// save the state hash before call executions
-	// (needed for block composition)
-	trace!("Getting hash of previous state ..");
-	let prev_state_hash = state.hash();
-	trace!("Loaded hash of previous state: {:?}", prev_state_hash);
+	let mut sidechain_db = load_initialized_state(&shard).map(SidechainDB::<SB::Block, _>::new)?;
```

Comment:
This looks good, but just to make sure - does this not work a little against making the sidechain optional? Or does it not matter here, because this function needs to be tidied up anyway?

Reply:
I doubt that this is a problem. The current STF design is a bigger problem IMO

Reply:
I guess I'll find out when I start working on #301. Maybe refactoring the STF is a hard requirement, before I can really solve #301 ?